nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -810,7 +810,8 @@ def forward(
             key_layer = self.inference_key_memory[:end, ...]
             value_layer = self.inference_value_memory[:end, ...]
             # Adjust attention mask
-            attention_mask = attention_mask[..., start:end, :end]
+            if end <= attention_mask.shape[-2]:
```

Comment:
why you assume `end > attention_mask.shape[-2]` can happen? if it happens, it should throw a exeception?

Reply:
also this logic is moved to the text generation changed by @ericharper for TE. @ericharper  please take a look.